Documentation/Components.md

```diff
@@ -0,0 +1,246 @@
+# Babylon Native Components
+
+This document gives a high-level overview of the components provided by 
+default in the Babylon Native repository. For information about the 
+definition of _component_ and the different categories of components, 
+please read the page on the 
+[Babylon Native build system](BuildSystem.md). For information about how to 
+create new components and add or reference them in Babylon Native, please
+read the page on 
+[extending Babylon Native](Extending.md).
+
+## Dependencies
+
+Components in this category encapsulate functionality brought in from 
+dependencies which are not logically a part of Babylon Native. Most of these
+are external libraries, and many of them are included as submodules.
+
+### arcana
+
+[Arcana](https://github.com/microsoft/arcana.cpp) is a collection of C++ 
+utilities. Most prominently, it includes an open-source and cross-platform 
+task and asynchrony system modelled after the 
+[PPL](https://docs.microsoft.com/en-us/cpp/parallel/concrt/parallel-patterns-library-ppl?view=vs-2019), 
+which Babylon Native makes extensive use of internally. Babylon Native also 
+makes internal use of several other utilities and data structures provided 
+by Arcana. However, no Arcana types are exposed by Babylon Native's APIs.
+
+### bgfx
+
+Babylon Native uses the 
+[bgfx](https://bkaradzic.github.io/bgfx/overview.html) family of 
+technologies internally as a graphics abstraction for cross-platform native
+rendering. Babylon Native consumes these dependencies through a 
+[fork](https://github.com/BabylonJS/bgfx.cmake) of the 
+[bgfx.cmake](https://github.com/widberg/bgfx.cmake) project, which provides
+an easy integration layer by which bgfx can be brought into Babylon 
+Native's CMake-based build system. The bgfx family of technologies are used
+internally by Babylon Native components that need a cross-platform 
+abstraction for GPU access and rendering work. However, no bgfx types are
+exposed by Babylon Native's APIs. 
+
+### glslang
+
+[glslang](https://github.com/KhronosGroup/glslang) is the reference compiler
+for GLSL provided by the Khronos Group. It is used internally by Babylon 
+Native as part of the 
+[shader transpilation pipeline](ShaderTranspilation.md).
+
+### gsl
+
+The [Guidelines Support Library](https://github.com/microsoft/gsl) is a 
+C++ utility library supplying several useful types and capabilities that
+are not supplied by the STL but are conceptually similar. It is used 
+internally throughout Babylon Native, but no GSL types are exposed by any
+Babylon Native API.
+
+### napi
+
+N-API is an integration layer that allows C and C++ native libraries to 
+interface with JavaScript engines in an engine-agnostic manner. Babylon
+Native's consumption of N-API is a heavily-modified spinoff of the 
```

Comment:
> heavily-modified spinoff

I'm not sure this is accurate. Maybe merge with the wording we have from the README.